File: src/Models/GarageC.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package Models;

import java.awt.Image;

/**
 *
 * @author helam
 */
public class GarageC {
   //attributs
    private int id_garage;
    private String nom_garage;
    private String adresse;
    private int numero;
    private int panne_moteur,pompe_a_eau,patin,essuie_glace,radiateur,ventilateur,duride,fuite_d_huile,vidange,filtre,batterie,amortisseur,frein_main,feu_d_eclairage;
    private String image;
    int taux_de_reduction;
    
    //contructeur

    public GarageC() {
    }

    public GarageC(int id_garage, String nom_garage, String adresse, int numero, int panne_moteur, int pompe_a_eau, int patin, int essuie_glace, int radiateur, int ventilateur, int duride, int fuite_d_huile, int vidange, int filtre, int batterie, int amortisseur, int frein_main, int feu_d_eclairage, int taux_de_reduction, String image) {
        this.id_garage = id_garage;
        this.nom_garage = nom_garage;
        this.adresse = adresse;
        this.numero = numero;
        this.panne_moteur = panne_moteur;
        this.pompe_a_eau = pompe_a_eau;
        this.patin = patin;
        this.essuie_glace = essuie_glace;
        this.radiateur = radiateur;
        this.ventilateur = ventilateur;
        this.duride = duride;
        this.fuite_d_huile = fuite_d_huile;
        this.vidange = vidange;
        this.filtre = filtre;
        this.batterie = batterie;
        this.amortisseur = amortisseur;
        this.frein_main = frein_main;
        this.feu_d_eclairage = feu_d_eclairage;
        this.image = image;
        this.taux_de_reduction = taux_de_reduction;
    }

    public GarageC(String nom_garage, String adresse, int numero, int panne_moteur, int pompe_a_eau, int patin, int essuie_glace, int radiateur, int ventilateur, int duride, int fuite_d_huile, int vidange, int filtre, int batterie, int amortisseur, int frein_main, int feu_d_eclairage, int taux_de_reduction,String image) {
        this.nom_garage = nom_garage;
        this.adresse = adresse;
        this.numero = numero;
        this.panne_moteur = panne_moteur;
        this.pompe_a_eau = pompe_a_eau;
        this.patin = patin;
        this.essuie_glace = essuie_glace;
        this.radiateur = radiateur;
        this.ventilateur = ventilateur;
        this.duride = duride;
        this.fuite_d_huile = fuite_d_huile;
        this.vidange = vidange;
        this.filtre = filtre;
        this.batterie = batterie;
        this.amortisseur = amortisseur;
        this.frein_main = frein_main;
        this.feu_d_eclairage = feu_d_eclairage;
        this.image = image;
        this.taux_de_reduction = taux_de_reduction;
    }

   
    
    //getters & setters

    public int getId_garage() {
        return id_garage;
    }

    public void setId_garage(int id_garage) {
        this.id_garage = id_garage;
    }

    public String getNom_garage() {
        return nom_garage;
    }

    public void setNom_garage(String nom_garage) {
        this.nom_garage = nom_garage;
    }

    public String getAdresse() {
        return adresse;
    }

    public void setAdresse(String adresse) {
        this.adresse = adresse;
    }

    public int getNumero() {
        return numero;
    }

    public void setNumero(int numero) {
        this.numero = numero;
    }

    public int getPanne_moteur() {
        return panne_moteur;
    }

    public void setPanne_moteur(int panne_moteur) {
        this.panne_moteur = panne_moteur;
    }

    public String getImage() {
        return image;
    }

    public void setImage(String image) {
        this.image = image;
    }

    public int getPompe_a_eau() {
        return pompe_a_eau;
    }

    public void setPompe_a_eau(int pompe_a_eau) {
        this.pompe_a_eau = pompe_a_eau;
    }

    public int getPatin() {
        return patin;
    }

    public void setPatin(int patin) {
        this.patin = patin;
    }

    public int getEssuie_glace() {
        return essuie_glace;
    }

    public void setEssuie_glace(int essuie_glace) {
        this.essuie_glace = essuie_glace;
    }

    public int getRadiateur() {
        return radiateur;
    }

    public void setRadiateur(int radiateur) {
        this.radiateur = radiateur;
    }

    public int getVentilateur() {
        return ventilateur;
    }

    public void setVentilateur(int ventilateur) {
        this.ventilateur = ventilateur;
    }

    public int getDuride() {
        return duride;
    }

    public void setDuride(int duride) {
        this.duride = duride;
    }

    public int getFuite_d_huile() {
        return fuite_d_huile;
    }

    public void setFuite_d_huile(int fuite_d_huile) {
        this.fuite_d_huile = fuite_d_huile;
    }

    public int getVidange() {
        return vidange;
    }

    public void setVidange(int vidange) {
        this.vidange = vidange;
    }

    public int getFiltre() {
        return filtre;
    }

    public void setFiltre(int filtre) {
        this.filtre = filtre;
    }

    public int getBatterie() {
        return batterie;
    }

    public void setBatterie(int batterie) {
        this.batterie = batterie;
    }

    public int getAmortisseur() {
        return amortisseur;
    }

    public void setAmortisseur(int amortisseur) {
        this.amortisseur = amortisseur;
    }

    public int getFrein_main() {
        return frein_main;
    }

    public void setFrein_main(int frein_main) {
        this.frein_main = frein_main;
    }

    public int getFeu_d_eclairage() {
        return feu_d_eclairage;
    }

    public void setFeu_d_eclairage(int feu_d_eclairage) {
        this.feu_d_eclairage = feu_d_eclairage;
    }

    public int getTaux_de_reduction() {
        return taux_de_reduction;
    }

    public void setTaux_de_reduction(int taux_de_reduction) {
        this.taux_de_reduction = taux_de_reduction;
    }

     
    
    ///toString 

    @Override
    public String toString() {
        return "GarageC{" + ", nom_garage=" + nom_garage + ", adresse=" + adresse + ", numero=" + numero + ", panne_moteur=" + panne_moteur + ", pompe_a_eau=" + pompe_a_eau + ", patin=" + patin + ", essuie_glace=" + essuie_glace + ", radiateur=" + radiateur + ", ventilateur=" + ventilateur + ", duride=" + duride + ", fuite_d_huile=" + fuite_d_huile + ", vidange=" + vidange + ", filtre=" + filtre + ", batterie=" + batterie + ", amortisseur=" + amortisseur + ", frein_main=" + frein_main + ", feu_d_eclairage=" + feu_d_eclairage + ", image=" + image + ", taux_de_reduction=" + taux_de_reduction + '}';
    }

   
   
    
}

File: src/Entity/Garagec.php
<?php

namespace App\Entity;

use App\Repository\GaragecRepository;
use Doctrine\ORM\Mapping as ORM;
use Symfony\Component\Validator\Constraints as Assert;
use Symfony\Component\Serializer\Annotation\Groups;

#[ORM\Entity(repositoryClass: GaragecRepository::class)]
class Garagec
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column(name:"id_garage")]
    #[Groups("garagec")]
    public ?int $id = null;

    #[ORM\Column(length: 255)]
    #[Assert\NotBlank(message:'ce champ est obligatoire')]
    #[Groups("garagec")]
    private ?string $nom_garage = null;

    #[ORM\Column(length: 255)]
    #[Groups("garagec")]
    //#[Assert\NotBlank(message:'ce champ est obligatoire veuillez selectionner un endoirt dans la map')]
    private ?string $adresse = null;

    #[ORM\Column]
    #[Assert\Length(
        min: 8,
        max: 8,
        minMessage: 'Your number name must be at least {{ limit }} characters long',
        maxMessage: 'Your first name cannot be longer than {{ limit }} characters',
    )]
    #[Groups("garagec")]
    private ?int $numero = null;

    #[ORM\Column]
    #[Assert\Positive(message:'le prix doit etre positive')]
    #[Assert\NotNull(message:'le champ ne doit pas etre vide')]
    #[Groups("garagec")]
    private ?int $panne_moteur = null;

    #[ORM\Column]
    #[Assert\Positive(message:'le prix doit etre positive')]
    #[Assert\NotNull(message:'le champ ne doit pas etre vide')]
    #[Groups("garagec")]
    private ?int $pompe_a_eau = null;

    #[ORM\Column]
    #[Assert\Positive(message:'le prix doit etre positive')]
    #[Assert\NotNull(message:'le champ ne doit pas etre vide')]
    #[Groups("garagec")]
    private ?int $patin = null;

    #[ORM\Column]
    #[Assert\Positive(message:'le prix doit etre positive')]
    #[Assert\NotNull(message:'le champ ne doit pas etre vide')]
    #[Groups("garagec")]
    private ?int $essuie_glace = null;

    #[ORM\Column]
    #[Assert\Positive(message:'le prix doit etre positive')]
    #[Assert\NotNull(message:'le champ ne doit pas etre vide')]
    #[Groups("garagec")]
    private ?int $radiateur = null;

    #[ORM\Column]
    #[Assert\Positive(message:'le prix doit etre positive')]
    #[Assert\NotNull(message:'le champ ne doit pas etre vide')]
    #[Groups("garagec")]
    private ?int $ventilateur = null;

    #[ORM\Column]
    #[Assert\Positive(message:'le prix doit etre positive')]
    #[Assert\NotNull(message:'le champ ne doit pas etre vide')]
    #[Groups("garagec")]
    private ?int $duride = null;

    #[ORM\Column]
    #[Assert\Positive(message:'le prix doit etre positive')]
    #[Assert\NotNull(message:'le champ ne doit pas etre vide')]
    #[Groups("garagec")]
    private ?int $fuite_d_huile = null;

    #[ORM\Column]
    #[Assert\Positive(message:'le prix doit etre positive')]
    #[Assert\NotNull(message:'le champ ne doit pas etre vide')]
    #[Groups("garagec")]
    private ?int $vidange = null;

    #[ORM\Column]
    #[Assert\Positive(message:'le prix doit etre positive')]
    #[Assert\NotNull(message:'le champ ne doit pas etre vide')]
    #[Groups("garagec")]
    private ?int $filtre = null;

    #[ORM\Column]
    #[Assert\Positive(message:'le prix doit etre positive')]
    #[Assert\NotNull(message:'le champ ne doit pas etre vide')]
    #[Groups("garagec")]
    private ?int $batterie = null;

    #[ORM\Column]
    #[Assert\Positive(message:'le prix doit etre positive')]
    #[Assert\NotNull(message:'le champ ne doit pas etre vide')]
    #[Groups("garagec")]
    private ?int $amortisseur = null;

    #[ORM\Column]
    #[Assert\Positive(message:'le prix doit etre positive')]
    #[Assert\NotNull(message:'le champ ne doit pas etre vide')]
    #[Groups("garagec")]
    private ?int $frein_main = null;

    #[ORM\Column]
    #[Assert\Positive(message:'le prix doit etre positive')]
    #[Assert\NotNull(message:'le champ ne doit pas etre vide')]
    #[Groups("garagec")]
    private ?int $feu_d_eclairage = null;

    #[ORM\Column]
    #[Assert\Positive(message:'le prix doit etre positive')]
    #[Assert\NotNull(message:'le champ ne doit pas etre vide')]
    #[Groups("garagec")]
    private ?int $taux_de_reduction = null;

    #[ORM\Column(length: 255)]
    #[Groups("garagec")]
    private ?string $image = null;

    public function getId(): ?int
    {
        return $this->id;
    }

    public function getNomGarage(): ?string
    {
        return $this->nom_garage;
    }

    public function setNomGarage(string $nom_garage): self
    {
        $this->nom_garage = $nom_garage;

        return $this;
    }

    public function getAdresse(): ?string
    {
        return $this->adresse;
    }

    public function setAdresse(string $adresse): self
    {
        $this->adresse = $adresse;

        return $this;
    }

    public function getNumero(): ?int
    {
        return $this->numero;
    }

    public function setNumero(int $numero): self
    {
        $this->numero = $numero;

        return $this;
    }

    public function getPanneMoteur(): ?int
    {
        return $this->panne_moteur;
    }

    public function setPanneMoteur(int $panne_moteur): self
    {
        $this->panne_moteur = $panne_moteur;

        return $this;
    }

    public function getPompeAEau(): ?int
    {
        return $this->pompe_a_eau;
    }

    public function setPompeAEau(int $pompe_a_eau): self
    {
        $this->pompe_a_eau = $pompe_a_eau;

        return $this;
    }

    public function getPatin(): ?int
    {
        return $this->patin;
    }

    public function setPatin(int $patin): self
    {
        $this->patin = $patin;

        return $this;
    }

    public function getEssuieGlace(): ?int
    {
        return $this->essuie_glace;
    }

    public function setEssuieGlace(int $essuie_glace): self
    {
        $this->essuie_glace = $essuie_glace;

        return $this;
    }

    public function getRadiateur(): ?int
    {
        return $this->radiateur;
    }

    public function setRadiateur(int $radiateur): self
    {
        $this->radiateur = $radiateur;

        return $this;
    }

    public function getVentilateur(): ?int
    {
        return $this->ventilateur;
    }

    public function setVentilateur(int $ventilateur): self
    {
        $this->ventilateur = $ventilateur;

        return $this;
    }

    public function getDuride(): ?int
    {
        return $this->duride;
    }

    public function setDuride(int $duride): self
    {
        $this->duride = $duride;

        return $this;
    }

    public function getFuiteDHuile(): ?int
    {
        return $this->fuite_d_huile;
    }

    public function setFuiteDHuile(int $fuite_d_huile): self
    {
        $this->fuite_d_huile = $fuite_d_huile;

        return $this;
    }

    public function getVidange(): ?int
    {
        return $this->vidange;
    }

    public function setVidange(int $vidange): self
    {
        $this->vidange = $vidange;

        return $this;
    }

    public function getFiltre(): ?int
    {
        return $this->filtre;
    }

    public function setFiltre(int $filtre): self
    {
        $this->filtre = $filtre;

        return $this;
    }

    public function getBatterie(): ?int
    {
        return $this->batterie;
    }

    public function setBatterie(int $batterie): self
    {
        $this->batterie = $batterie;

        return $this;
    }

    public function getAmortisseur(): ?int
    {
        return $this->amortisseur;
    }

    public function setAmortisseur(int $amortisseur): self
    {
        $this->amortisseur = $amortisseur;

        return $this;
    }

    public function getFreinMain(): ?int
    {
        return $this->frein_main;
    }

    public function setFreinMain(int $frein_main): self
    {
        $this->frein_main = $frein_main;

        return $this;
    }

    public function getFeuDEclairage(): ?int
    {
        return $this->feu_d_eclairage;
    }

    public function setFeuDEclairage(int $feu_d_eclairage): self
    {
        $this->feu_d_eclairage = $feu_d_eclairage;

        return $this;
    }

    public function getTauxDeReduction(): ?int
    {
        return $this->taux_de_reduction;
    }

    public function setTauxDeReduction(int $taux_de_reduction): self
    {
        $this->taux_de_reduction = $taux_de_reduction;

        return $this;
    }

    public function getImage(): ?string
    {
        return $this->image;
    }

    public function setImage(string $image): self
    {
        $this->image = $image;

        return $this;
    }
}
